server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -754,59 +757,90 @@ public GetResult get(
         try (ReleasableLock ignored = readLock.acquire()) {
             ensureOpen();
             if (get.realtime()) {
-                final VersionValue versionValue;
```

Comment:
All I've done here is to break this down into two pieces. Alternative would be probably a flag to shortcut this, but I'm not sure it makes this more readable. Open for suggestions.